common/lib/xmodule/xmodule/css/capa/display.scss

```diff
@@ -906,47 +905,79 @@ div.problem {
 // +Problem - Actions
 // ====================
 div.problem .action {
-  margin-top: $baseline;
+  @include margin($baseline 0 $baseline 0);
+  min-height: $baseline;
 
-  .save, .check, .show, .reset, .hint-button {
-    @include margin-right($baseline/2);
-    margin-bottom: ($baseline/2);
-    height: ($baseline*2);
-    vertical-align: middle;
-    text-transform: uppercase;
-    font-weight: 600;
-    @media print {
-      display: none;
+  // override default button hover
+  button {
+    &:hover {
+      background-image: none;
+      box-shadow: none;
+    }
+    &:focus {
+      box-shadow: none;
+    }
+    &.btn-default {
+      background-color: transparent;
+    }
+    &.btn-brand {
+      &:hover {
+        background-color: #0ea6ec;
+      }
+      &.check {
+        &.is-disabled, &.disabled {
+          background: transparent;
+          border-color: #d2d0d0;
+          color: #6b6969;
+        }
+      }
     }
   }
 
-  .save {
-    @extend .blue-button !optional;
+  @media (min-width: 768px) {
+
+    .problem-action-buttons-wrapper {
+      position: absolute;
+      @include margin-right(($baseline*.75));
+      right: 0;
+    }
+
   }
 
-  .show {
+  .problem-action-button-wrapper {
+    @include padding(($baseline*.75) 0 ($baseline/2) ($baseline/4));
+    border-right: 1px solid #ddd;
+    &:last-child {
+      border: none;
+    }
+  }
 
-    .show-label {
-      font-weight: 600;
-      font-size: 1.0em;
+  .save, .show, .reset, .hint-button {
+    margin-bottom: ($baseline/2);
+    vertical-align: middle;
+    @media print {
+      display: none;
+    }
+    .icon {
+      display: block;
     }
   }
 
   .submission_feedback {
-    // background: #F3F3F3;
-    // border: 1px solid #ddd;
-    // border-radius: 3px;
-    // padding: 8px 12px;
-    // margin-top: ($baseline/2);
     @include margin-left($baseline/2);
+    height: ($baseline*2);
     display: inline-block;
-    margin-top: 8px;
     color: $gray-d1;
-    font-style: italic;
+    font-size: 0.9em;;
     -webkit-font-smoothing: antialiased;
+
+    @media (min-width: 768px) {
+      font-size: 0.8em;
```

Comment:
Consider using `rem`s instead of explicitly changing font size based on screen width.

(No idea if that's actually feasible given the existing CSS/markup in this area of the codebase, but I thought I'd mention it.)